### DISCUSSION TITLE
Unable to do a request with filter on mysql db with join

### DISCUSSION THREAD

**Discussion by wasims1** [directus/directus#14130] [Archive]
Hi, we have a DB with two tables with one-to-many relationship. We have a directus instance that connects to the DB. The relationship between the two tables was not initiated through directus but rather through our app nodejs backend. We need to query these two tables with filters for both of them using directus. However, we have been unable to figure out how to structure the api query as it seems that directus doesn't `see` this one-to-many relationship...

Any solutions are appreciated...

**Comment by br41nslug** [comment]
Directus can only traverse relations of which it is aware. There is currently no way to join two arbitrary collections without creating a relation between them first. so the solution would be to create the o2m field via directus (or at least add it to `directus_relations` and `directus_fields` manually so Directus is aware of its existence)

**Comment by rijkvanzanten** [comment]
Directus relies on foreign key constraints to be aware of relationships in the database. Make sure you configure a foreign key constraint when creating relationships in the database directly 👍🏻

**Reply by wasims1** [reply]
thanks for the reply... I do have a foreign key constraint in the database, however it was created outside of direcuts... just to be clear, do i still need to add the relationship in directus itself for it to be aware of it?

**Reply by rijkvanzanten** [reply]
Directus will introspect the foreign key constraints from the database, and treat them as relational fields, so that should be enough
